File: Data Structures & Algorithms/longest-common-prefix/submission-0.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string}
     */
    longestCommonPrefix(strs) {
        let ans="",cnt=strs[0].length;
        for(let i=1;i<strs.length;i++){
            for(let j=0;j<cnt;j++){
                if(strs[0][j]!=strs[i][j]){
                    cnt=j;
                }
            }
        }
        if(cnt==0)return ans;
        ans=strs[0].slice(0,cnt);
        return ans;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mp;
        string key;
        for (auto& s : strs) {
            key = getKey(s);
            mp[key].push_back(s);
        }
        vector<vector<string>> ans;
        for (auto& it : mp) {
            ans.push_back(it.second);
        }
        return ans;
    }
    string getKey(string s) {
        vector<int> v(26, 0);
        for (auto& i : s) {
            v[i - 'a']++;
        }
        string key = to_string(v[0]) + ",";
        for (int i = 1; i < 26; i++) {
            key += to_string(v[i]) + ",";
        }
        return key;
    }
};
